test/form/samples/undefined-var/main.js

```diff
@@ -0,0 +1,10 @@
+var x;
+var y = undefined;
+var z;
+if (x ? (console.log('a'), false) : (console.log('b'), true))
+	 console.log('yes');
+if (y ? (console.log('a'), false) : (console.log('b'), false))
+	console.log('no');
+if (z ? (console.log('a'), false) : (console.log('b'), false))
+	console.log('yes');
+z = 1;
```

Comment:
I'm not too happy the comma expression logs. They sure prevent the ternary operators from being tree-shaken but in the light of what you want to test, I would rather have simple direct tests such as

```js
if (x) console.log('no');
if (!x) console.log('yes');
```

i.e. six simple test cases which are more readable and where it is easily clear what went wrong when the test turns red. It's also easy to overlook in your test that the second value in the comma expressions are not symmetric.